src/storage/src/table/storage_table.rs

```diff
@@ -380,16 +380,34 @@ impl<S: StateStore, RS: RowSerde, const T: AccessType> StorageTableBase<S, RS, T
     pub async fn get_row(&self, pk: &Row, epoch: u64) -> StorageResult<Option<Row>> {
         // TODO: use multi-get for storage get_row
         let serialized_pk = self.serialize_pk_with_vnode(pk);
-
-        let sentinel_key =
-            serialize_pk_and_column_id(&serialized_pk, &SENTINEL_CELL_ID).map_err(err)?;
-        if self.keyspace.get(&sentinel_key, epoch).await?.is_none() {
-            // if sentinel cell is none, this row doesn't exist
-            return Ok(None);
-        };
-
         let data_types = self.schema().data_types();
         let mut deserializer = RS::create_deserializer(self.mapping.clone(), data_types);
+        let sentinel_key = <RS as RowSerde>::Serializer::serialize_sentinel_cell(
+            &serialized_pk,
+            &SENTINEL_CELL_ID,
+        )
+        .map_err(err)?;
+        match sentinel_key {
+            Some(sentinel_key) => {
+                if self.keyspace.get(&sentinel_key, epoch).await?.is_none() {
+                    // if sentinel cell is none, this row doesn't exist
+                    return Ok(None);
+                };
+            }
+            // if sentinel cell does not exist, the encoding format is row-based.
+            None => {
```

Comment:
This part is not generic enough, but may be we can tolerate it as we will remove cell-based encoding later.